pkg/controller/datadogagentdeployment/agent.go

```diff
@@ -352,3 +350,34 @@ func newDaemonsetObjectMetaData(agentdeployment *datadoghqv1alpha1.DatadogAgentD
 		Annotations: annotations,
 	}
 }
+
+func getAgentCustomConfigConfigMapName(dad *datadoghqv1alpha1.DatadogAgentDeployment) string {
+	return fmt.Sprintf("%s-datadog-yaml", dad.Name)
+}
+
+func buildAgentConfigurationConfigMap(dad *datadoghqv1alpha1.DatadogAgentDeployment) (*corev1.ConfigMap, error) {
+	if dad.Spec.Agent.CustomConfig == nil {
+		return nil, nil
+	}
+
+	// Validate that user input is valid YAML
+	// Maybe later we can implement that directly verifies against Agent configuration?
```

Comment:
It's probably worth filing an issue for this right away.

Reply:
That is something I'm wondering about. It will for sure require a dependency (maybe not code, but still) on the agent, and as we can deploy "any" version of the agent with the operator, it might not be easy, so I'm not sure if we really want to do it or not